webviz_config/testing/_composite.py

```diff
@@ -0,0 +1,95 @@
+import pathlib
+from selenium.webdriver.common.by import By
+import time
+
+from dash.testing.composite import Browser
+import dash
+from webviz_config import WebvizSettings
+from webviz_config.common_cache import CACHE
+from webviz_config.themes import default_theme
+from webviz_config.webviz_factory_registry import WEBVIZ_FACTORY_REGISTRY
+from webviz_config.webviz_instance_info import WEBVIZ_INSTANCE_INFO, WebvizRunMode
+
+import webviz_core_components as wcc
+
+from ._webviz_ids import WebvizIds
+
+
+class WebvizComposite(Browser):
+    def __init__(self, server, **kwargs):
+        super().__init__(**kwargs)
+        self.app = dash.Dash(__name__)
+        self.server = server
+        self.init_app()
+        self.plugin = None
+
+    def init_app(self):
+        WEBVIZ_INSTANCE_INFO.initialize(
+            dash_app=self.app,
+            run_mode=WebvizRunMode.NON_PORTABLE,
+            theme=default_theme,
+            storage_folder=pathlib.Path(__file__).resolve().parent,
+        )
+        try:
+            WEBVIZ_FACTORY_REGISTRY.initialize(None)
+        except RuntimeError:
+            pass
+
+        self.app.css.config.serve_locally = True
+        self.app.scripts.config.serve_locally = True
+        self.app.config.suppress_callback_exceptions = True
+        CACHE.init_app(self.app.server)
+
+    def start_server(self, plugin, **kwargs):
+        """Start the local server with app."""
+
+        self.app.layout = dash.html.Div(
+            className=WebvizIds.LAYOUT_WRAPPER,
+            children=[
+                wcc.WebvizContentManager(
+                    id=WebvizIds.CONTENT_MANAGER,
+                    children=[
+                        wcc.WebvizSettingsDrawer(
+                            id=WebvizIds.SETTINGS_DRAWER,
+                            children=plugin.get_all_settings(),
+                        ),
+                        wcc.WebvizPluginsWrapper(
+                            id=WebvizIds.PLUGINS_WRAPPER,
+                            children=plugin.plugin_layout(),
+                        ),
+                    ],
+                ),
+            ],
+        )
+        self.plugin = plugin
+        # start server with app and pass Dash arguments
+        self.server(self.app, **kwargs)
+
+        # set the default server_url, it implicitly call wait_for_page
+        self.server_url = self.server.url
+
+    def toggle_webviz_drawer(self):
+        """Open the plugin settings drawer"""
+        self.find_element(WebvizIds.SETTINGS_DRAWER_TOGGLE_OPEN).click()
+        time.sleep(0.1)
```

Comment:
I think selenium has API for doing "wait until this elements is visible". Then we ensure we always wait enough, but not more. Should we try to use that instead of `time.sleep`?